File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = "AEIOUaeiou"
        tosort = []
        result = [0]*len(s)
        for i in range(len(s)):
            if s[i] in vowels:
                tosort.append(s[i])
            else:
                result[i] = s[i]
        tosort.sort()
        for i in range(len(s)):
            if result[i] == 0:
                result[i] = tosort.pop(0)
        return ''.join(result)

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        char[] arr = s.toCharArray();
        String str = new String("AEIOUaeiou");
        List<Character> vowels = new ArrayList<Character>();
        for(int i = 0; i < arr.length ; i++){
            if(str.indexOf(arr[i]) != -1){
                vowels.add(arr[i]);
                arr[i] = '.';
            }
        }
        Collections.sort(vowels);
        for(int i = 0; i < arr.length ; i++){
            if(arr[i] == '.'){
                arr[i] = vowels.get(0);
                vowels.remove(0);
            }
        }
        return String.valueOf(arr);
    }
}